frontend/src/utils/helpers/groupByProperty.ts

```diff
@@ -0,0 +1,12 @@
+const groupByProperty = <T>(array: T[], propertyKey: keyof T): Map<T[keyof T], T[]> => {
+  return array.reduce((group: Map<T[keyof T], T[]>, item: T) => {
+    const key = item[propertyKey];
+    if (!group.has(key)) {

```

Comment:
what is **key & item** ? Could u add more clear naming?

Reply:
it's depesonolized function so it'sbetter also use depersonolized names for that